### PR TITLE
Drop support for iOS < 15.7

### PR DESCRIPTION
**Pull request by davidtaylorhq**
https://meta.discourse.org/t/224747

### CODE REVIEW

File: app/assets/javascripts/discourse/config/targets.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ const browsers = [
 
 if (isCI || isProduction) {
   // https://meta.discourse.org/t/224747
-  browsers.push("Safari 12");
+  browsers.push("Safari 14");
```
Comment (xfalcox) [inline app/assets/javascripts/discourse/config/targets.js:14]:
```suggestion
  browsers.push("Safari 15");
```

File: spec/lib/discourse_js_processor_spec.rb [hunk 2]
```diff
@@ -37,6 +37,45 @@
...
+      optional?.chaining;
+      const template = func`test`;
+      class MyClass {
+        classProperty = 1;
+        #privateProperty = 1;
+        #privateMethod() {
+          console.log("hello world");
+        }
+      }
+      let numbericSeparator = 100_000_000;
```
Comment (CvX) [inline spec/lib/discourse_js_processor_spec.rb:51]:
```suggestion
      let numericSeparator = 100_000_000;
```

File: spec/lib/mobile_detection_spec.rb [hunk 3]
```diff
@@ -36,11 +36,13 @@
...
     modern_user_agents.each do |agent|
-      expect(MobileDetection.modern_mobile_device?(agent)).to eq(true)
+      expect(MobileDetection.modern_mobile_device?(agent)).to eq(false),
+      "Failed User Agent: '#{agent}'"
     end
 
     old_user_agents.each do |agent|
-      expect(MobileDetection.modern_mobile_device?(agent)).to eq(false)
+      expect(MobileDetection.modern_mobile_device?(agent)).to eq(false),
+      "Failed User Agent: '#{agent}'"
```
Comment (nbianca) [inline spec/lib/mobile_detection_spec.rb:45]: The lack of indentation here is strange and a bit misleading.

File: spec/lib/discourse_js_processor_spec.rb (hunk 2 above)
Comment (nbianca) [reply spec/lib/discourse_js_processor_spec.rb:51]: I believe this was not addressed.

**Review by nbianca — APPROVED** [review]

File: spec/lib/mobile_detection_spec.rb [hunk 4]
```diff
@@ -36,11 +36,13 @@
 
   it "detects modern browsers correctly" do
     modern_user_agents.each do |agent|
-      expect(MobileDetection.modern_mobile_device?(agent)).to eq(true)
+      expect(MobileDetection.modern_mobile_device?(agent)).to eq(false),
+      "Failed User Agent: '#{agent}'"
```
[the remark this answers is not in the document]
Comment (davidtaylorhq) [reply spec/lib/mobile_detection_spec.rb]: True - this is how syntax_tree chose to lay it out. I've added some parenthesis which should make it clearer

File: spec/lib/discourse_js_processor_spec.rb (hunk 2 above)
Comment (davidtaylorhq) [reply spec/lib/discourse_js_processor_spec.rb]: Right 🤦‍♂️. Fixed now - thanks @nbianca

File: spec/lib/mobile_detection_spec.rb (hunk 3 above)
Comment (davidtaylorhq) [reply spec/lib/mobile_detection_spec.rb]: Added parenthesis

**Pull request merged**